pkg/controller/job/job_controller.go

```diff
@@ -772,6 +766,8 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 
 	// if job was finished previously, we don't want to redo the termination
 	if util.IsJobFinished(&job) {
+		// The job shouldn't be marked as finished until all pod finalizers are removed.
```

Comment:
```suggestion
		// The job shouldn't be marked as finished until all pod finalizers are removed.
		// Cleaning pod finalizers one more time just in case.
```